overlord/ifacestate/hotplug_test.go

```diff
@@ -1120,14 +1120,14 @@ func (s *hotplugSuite) TestRemoveDeviceTasks(c *C) {
 	iface, key, err := ifacestate.GetHotplugAttrs(task1)
 	c.Assert(err, IsNil)
 	c.Assert(iface, Equals, "interface")
-	c.Assert(key, Equals, "key")
+	c.Assert(key, DeepEquals, snap.HotplugKey("key"))
 
 	task2 := tss.Tasks()[1]
 	c.Assert(task2.Kind(), Equals, "hotplug-remove-slot")
 	iface, key, err = ifacestate.GetHotplugAttrs(task2)
 	c.Assert(err, IsNil)
 	c.Assert(iface, Equals, "interface")
-	c.Assert(key, Equals, "key")
+	c.Assert(key, DeepEquals, snap.HotplugKey("key"))
```

Comment:
Why does this now require deep equals? Is this due to the fact that the string is wrapped in a new type?

Reply:
It doesn't require DeepEquals, I just felt it's better for custom types (in particular if they ever grow up, probably unlikely here). I can revert back to Equals if you think that's confusing?